pkg/sql/crdb_internal.go

```diff
@@ -340,6 +341,62 @@ CREATE TABLE crdb_internal.tables (
 	},
 }
 
+var crdbInternalTablesTableLastStats = virtualSchemaTable{
+	comment: "the latest stats for all tables accessible by current user in current database (KV scan)",
+	schema: `
+CREATE TABLE crdb_internal.table_row_statistics (
+  table_id                   INT         NOT NULL,
+  table_name                 STRING      NOT NULL,
+  estimated_row_count        INT
+)`,
+	populate: func(ctx context.Context, p *planner, db *sqlbase.ImmutableDatabaseDescriptor, addRow func(...tree.Datum) error) error {
+		// Collect the latests statistics for all tables.
+		query := `
+           SELECT s."tableID", max(s."rowCount")
+             FROM system.table_statistics AS s
+             JOIN (
+                    SELECT "tableID", max("createdAt") AS last_dt
+                      FROM system.table_statistics
+                     GROUP BY "tableID"
+                  ) AS l ON l."tableID" = s."tableID" AND l.last_dt = s."createdAt"
+            GROUP BY s."tableID"`
+		statRows, err := p.ExtendedEvalContext().ExecCfg.InternalExecutor.QueryEx(
+			ctx, "crdb-internal-statistics-table", p.txn,
+			sqlbase.InternalExecutorSessionDataOverride{User: security.RootUser},
+			query)
+		if err != nil {
+			return err
+		}
+
+		// Convert statistics into map: tableID -> rowCount.
+		statMap := make(map[tree.DInt]tree.Datum)
+		for _, r := range statRows {
+			statMap[tree.MustBeDInt(r[0])] = r[1]
+		}
+
+		// Walk over all available tables and show row count for each of them
+		// using collected statistics.
+		return forEachTableDescAll(ctx, p, db, virtualMany,
+			func(db *sqlbase.ImmutableDatabaseDescriptor, _ string, table *ImmutableTableDescriptor) error {
+				tableID := tree.DInt(table.ID)
+				rowCount := tree.DNull
+				// For Virtual Tables report NULL row count.
+				if !table.IsVirtualTable() {
```

Comment:
Do virtual tables show up in the stats? I think we also want to just default row count to NULL if we cannot find the table in statMap

Reply:
Yep, virtual tables will show up in `crdb_internal.table_row_statistics` (even if there's no any rows for it in `system.table_statistics`).
All virtual tables will have NULL number of rows as you asked here:

- https://github.com/cockroachdb/cockroach/pull/52203#pullrequestreview-461284517

> Also, this row count is irrelevant for virtual tables, so I would prefer if we reported null rather than 0 for them.